ui/src/app/cohort-search/attributes-page-v2/attributes-page-v2.component.tsx

```diff
@@ -0,0 +1,610 @@
+  },
+  errorItem: {
+    lineHeight: '16px',
+  },
+});
+
+const optionUtil = {
+  ANY: {display: 'Any value', code: 'Any'},
+  EQUAL: {display: '= ', code: '01'},
+  GREATER_THAN_OR_EQUAL_TO: {display: '>= ', code: '02'},
+  LESS_THAN_OR_EQUAL_TO: {display: '<= ', code: '03'},
+  BETWEEN: {display: '', code: '04'},
+};
+
+interface AttributeForm {
+  exists: boolean; // Check if attribute values exist (Measurements only)
+  num: Array<any>; // Numerical attributes (Physical Measurements or Measurements)
+  cat: Array<any>; // Categorical attributes (Measurements only)
+}
+
+interface Props {
+  close: Function;
+  criteria: Array<Selection>;
+  node: any;
+  workspace: WorkspaceData;
+}
+
+interface State {
+  calculating: boolean;
+  count: number;
+  countError: boolean;
+  form: AttributeForm;
+  loading: boolean;
+  options: any;
+}
+export const AttributesPageV2 = fp.flow(withCurrentWorkspace(), withCurrentCohortCriteria()) (
+  class extends React.Component<Props, State> {
+    constructor(props: Props) {
+      super(props);
+      this.state = {
+        calculating: false,
+        count: null,
+        countError: false,
+        form: {exists: false, num: [], cat: []},
+        loading: true,
+        options: [
+          {label: 'Equals', value: Operator.EQUAL},
+          {label: 'Greater Than or Equal To', value: Operator.GREATERTHANOREQUALTO},
+          {label: 'Less Than or Equal To', value: Operator.LESSTHANOREQUALTO},
+          {label: 'Between', value: Operator.BETWEEN},
+        ],
+      };
+    }
+
+    componentDidMount() {
+      const{options} = this.state;
+      if (!this.isMeasurement) {
+        options.unshift({label: optionUtil.ANY.display, value: AttrName[AttrName.ANY]});
+        this.setState({options}, () => this.initAttributeForm());
+      } else {
+        this.initAttributeForm();
+      }
+    }
+
+    componentDidUpdate(prevProps: Readonly<Props>): void {
+      if (this.props.node !== prevProps.node) {
+        // A different node has been selected, so we reset the form and load the new attributes
+        this.setState({
+          form: {exists: false, num: [], cat: []},
+          loading: true
+        }, () => this.initAttributeForm());
+      }
+    }
+
+    initAttributeForm() {
+      const {node: {subtype}} = this.props;
+      const{form} = this.state;
+      if (this.hasRange) {
+        this.getAttributes();
+      } else {
+        form.num = subtype === CriteriaSubType[CriteriaSubType.BP]
+          ? JSON.parse(JSON.stringify(PREDEFINED_ATTRIBUTES.BP_DETAIL))
+          : [{name: subtype, operator: 'ANY', operands: []}];
+        this.setState({form, count: this.nodeCount, loading: false});
+      }
+    }
+
+    getAttributes() {
+      const {node: {conceptId}} = this.props;
+      const{form} = this.state;
+      const {cdrVersionId} = currentWorkspaceStore.getValue();
+      cohortBuilderApi().findCriteriaAttributeByConceptId(+cdrVersionId, conceptId).then(resp => {
```

Comment:
interesting whats the + for?

Reply:
Just converts `cdrVersionId` from a string to a number for the API call.